src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -248,7 +245,7 @@ private static bool TryGetProjectDisplayName(IVsHierarchy hierarchy, out string
 
         public string ProjectType => _projectType;
 
-        public Workspace Workspace => (Workspace)_visualStudioWorkspaceOpt ?? _miscellaneousFilesWorkspaceOpt;
+        public Workspace Workspace => _visualStudioWorkspaceOpt;
```

Comment:
This sounds dangerous (returning an optional value).

This should probably be renamed or removed.